### PR TITLE
Refresh history data when page is changed

### PR DESCRIPTION
**Pull request by racheljay**
This PR solves issue #3446 where the same data was persisting for different tests. I solved this by checking for and clearing any existing data before running the query for new data. I also added logic to hide the history timelines when a new test page is visited after viewing a previous one.

### CODE REVIEW

File: webapp/components/new-test-results-history-grid.js [hunk 1]
```diff
@@ -88,21 +88,24 @@ class TestResultsGrid extends PolymerElement {
...
 
-  displayCharts(showTestHistory) {
-    if (!showTestHistory || this.historicalData !== undefined) {
+  displayCharts(showTestHistory, path) {
+    if (!path) {
+      return;
+    }
+    if (!showTestHistory) {
       return;
     }
```
Comment (DanielRyanSmith) [inline webapp/components/new-test-results-history-grid.js:105]:
```suggestion
    if (!path || !showTestHistory) {
      return;
    }
```

File: webapp/components/new-test-results-history-grid.js [hunk 2]
```diff
@@ -231,17 +234,22 @@ class TestResultsGrid extends PolymerElement {
   }
 
   // get test history and aligned run data
-  async getTestHistory() {
-    if (!this.path) {
+  async getTestHistory(path) {
+    if (!path) {
       throw new Error('Test path is undefined');
     }
```
Comment (DanielRyanSmith) [inline webapp/components/new-test-results-history-grid.js:240]: This check is redundant now that the value is checked for in `displayCharts()`, so this block can probably just be removed

**Review by DanielRyanSmith — APPROVED** [review]
LGTM with some nits

File: webapp/components/new-test-results-history-grid.js (hunk 1 above)
Comment (racheljay) [reply webapp/components/new-test-results-history-grid.js]: done 👍

File: webapp/components/new-test-results-history-grid.js (hunk 2 above)
Comment (racheljay) [reply webapp/components/new-test-results-history-grid.js]: done 👍

File: webapp/components/new-test-results-history-grid.js [hunk 3]
```diff
@@ -88,21 +88,22 @@ class TestResultsGrid extends PolymerElement {
...
   }
 
   static get is() {
     return 'new-test-results-history-grid';
   }
 
-  displayCharts(showTestHistory) {
-    if (!showTestHistory || this.historicalData !== undefined) {
+  displayCharts(showTestHistory, path) {
+    if (!path || !showTestHistory) {
```
Comment (KyleJu) [inline webapp/components/new-test-results-history-grid.js:100]: We should only reload data when it is a subtest path

File: webapp/components/new-test-results-history-grid.js [hunk 4]
```diff
@@ -88,21 +88,22 @@ class TestResultsGrid extends PolymerElement {
...
   }
 
   static get is() {
     return 'new-test-results-history-grid';
   }
 
-  displayCharts(showTestHistory) {
-    if (!showTestHistory || this.historicalData !== undefined) {
+  displayCharts(showTestHistory, path) {
+    if (!path || !showTestHistory || path.split('?')[0].slice(-5) !== ".html") {
```
Comment (racheljay) [inline webapp/components/new-test-results-history-grid.js:100]: @KyleJu I believe this is now accurately checking for subtests urls. Can you let me know if there are any edge cases you can think of that I'm not covering?

Comment (KyleJu) [reply webapp/components/new-test-results-history-grid.js:100]: Could you use this method for checking instead? https://github.com/web-platform-tests/wpt.fyi/blob/7d1370ea05466fc84533ebed782d348756e102d5/webapp/components/path.js#L70

Comment (racheljay) [reply webapp/components/new-test-results-history-grid.js]: Done 👍

**Review by KyleJu — APPROVED** [review]
